### PR TITLE
ACP 6.3 SOF support and Add ACP SRAM option for Data bin loading

### PR DESCRIPTION
**Pull request by vijendarmukunda**
This patch series contains changes for adding SOF support for ACP v6.3 and ACP SRAM option to load the Data bin contents.

Pull request for ACP FW changes has been created.
https://github.com/thesofproject/sof/pull/8335

### CODE REVIEW

File: sound/soc/amd/acp-config.c [hunk 1]
```diff
@@ -263,4 +263,16 @@ struct snd_soc_acpi_mach snd_soc_acpi_amd_rmb_sof_machines[] = {
 };
 EXPORT_SYMBOL(snd_soc_acpi_amd_rmb_sof_machines);
 
+struct snd_soc_acpi_mach snd_soc_acpi_amd_acp63_sof_machines[] = {
+	{
+		.id = "AMDI1019",
+		.drv_name = "acp63-dsp",
+		.pdata = &acp_quirk_data,
+		.fw_filename = "sof-acp_6_3.ri",
+		.sof_tplg_filename = "sof-acp_6_3.tplg",
```
Comment (plbossart) [inline sound/soc/amd/acp-config.c:272]: nit-pick: the use of _ is inconsistent, you used - everywhere so far.

File: sound/soc/sof/amd/acp.h [hunk 2]
```diff
@@ -84,7 +84,7 @@
 #define EXCEPT_MAX_HDR_SIZE			0x400
 #define AMD_STACK_DUMP_SIZE			32
 
-#define SRAM1_SIZE				0x13A000
+#define SRAM1_SIZE				0x280000
```
Comment (plbossart) [inline sound/soc/sof/amd/acp.h:87]: does this work for older platforms?

File: sound/soc/sof/amd/acp-loader.c [hunk 3]
```diff
@@ -83,11 +83,21 @@ int acp_dsp_block_write(struct snd_sof_dev *sdev, enum snd_sof_fw_blk_type blk_t
 		}
 		dest = adata->data_buf + offset;
 		adata->fw_data_bin_size = size + offset;
+		adata->is_dram_in_use = true;
```
Comment (plbossart) [inline sound/soc/sof/amd/acp-loader.c:86]: I would split this patch in two with first the DRAM usage (no functionality change) and then the SRAM parts.

File: sound/soc/sof/amd/acp.h (hunk 2 above)
Comment (vijendarmukunda) [reply sound/soc/sof/amd/acp.h:87]: Yes. It works.

File: sound/soc/sof/amd/acp-loader.c (hunk 3 above)
Comment (vijendarmukunda) [reply sound/soc/sof/amd/acp-loader.c:86]: will split the patch series

Comment (vijendarmukunda) [reply sound/soc/sof/amd/acp-loader.c:86]: done

**Review by plbossart — APPROVED** [review]

**Review by bhiregoudar — APPROVED** [review]

**Pull request merged**